### ISSUE TITLE
Check if TV is switched off using Domoticz

### ISSUE THREAD

**Issue by gerard33** [gerard33/sony-bravia#7]
Check in heartbeat function if TV has just been switched off using Domoticz.
In that case don't start GetTVInfo, otherwise the error below could occur, because the heartbeat function is started so shortly after switching the TV off, it wont notice it's off yet.

```
2017-05-08 22:27:48.899 User: Admin initiated a switch command (41/Sony TV - Status/Off)
2017-05-08 22:27:48.934 (Sony TV) onCommand called for Unit 1: Parameter 'Off', Level: 0
2017-05-08 22:27:49.072 (Sony TV) Update 0:'Off' (Sony TV - Status)
2017-05-08 22:27:49.073 (Sony TV) Update 0:'10' (Sony TV - Volume)
2017-05-08 22:27:49.075 (Sony TV) Update 0:'0' (Sony TV - Source)
2017-05-08 22:27:49.076 (Sony TV) Update 0:'0' (Sony TV - Control)

2017-05-08 22:27:49.077 (Sony TV) Update 0:'0' (Sony TV - Channel)
2017-05-08 22:27:49.853 (Sony TV) Update 1:'NPO 3 HD (003) - Trippers [21:50 - 22:35]' (Sony TV - Status)
2017-05-08 22:27:49.854 (Sony TV) Update 1:'10' (Sony TV - Source)
2017-05-08 22:27:51.251 Error: (Sony TV) JSON request error:{
"id": 1,
"error": [
7,
"not power-on"
]
}
2017-05-08 22:27:51.252 Error: (Sony TV) 'onHeartbeat' failed 'TypeError':''NoneType' object is not subscriptable'.
2017-05-08 22:27:51.252 Error: (Sony TV) ----> Line 411 in /home/gerard/domoticz/plugins/sony/plugin.py, function onHeartbeat
2017-05-08 22:27:51.252 Error: (Sony TV) ----> Line 298 in /home/gerard/domoticz/plugins/sony/plugin.py, function onHeartbeat
2017-05-08 22:27:51.252 Error: (Sony TV) ----> Line 384 in /home/gerard/domoticz/plugins/sony/plugin.py, function GetTVInfo
2017-05-08 22:27:59.719 (Sony TV) Update 0:'Off' (Sony TV - Status)
2017-05-08 22:27:59.720 (Sony TV) Update 0:'None' (Sony TV - Volume)
2017-05-08 22:27:59.721 (Sony TV) Update 0:'0' (Sony TV - Source)
```

**Comment by gerard33** [comment]
Try/except added to the Heartbeat function.

```
if tvStatus == 'active':                        # TV is on
    self.powerOn = True
    try:
        self.GetTVInfo()
    except Exception as err:
        Domoticz.Error('No data received from TV, probably it has just been turned off (' +  err + ')')
```

**Comment by gerard33** [comment]
Changed except to `except (Exception, TypeError) as err:`.
Issue seems to be solved. Closing this one.

**Comment by gerard33** [comment]
The right solution to fix this issue is part of c8a1787.
```
if tvStatus == 'active':                        # TV is on
            self.powerOn = True
            try:
                self.GetTVInfo()
            except Exception as err:
                Domoticz.Error("No data received from TV, probably it has just been turned off (" +  str(err) + ")")
```
